services/swagger.js

```diff
@@ -151,20 +151,20 @@ function flattenModelDefinitions(definitions) {
     });
 }
 
-function flattenModel(item, props) {
+function mergeToFlatModel(flatModel, model) {
     //have to convert 'required' array to object for merging purposes
-    if (item.required && Array.isArray(item.required)) {
-        var requiredObj = {};
-        item.required.reduce(function (prevVal, currentVal, idx) {
-            prevVal[currentVal] = idx;
-            return prevVal;
-        }, requiredObj);
-        item.required = requiredObj;
+    if (model.required && Array.isArray(model.required)) {
+        var requiredAsObj = {};
```

Comment:
is this variable necessary?
can't the result of the reduce be assigned directly to `model.required`?